semantic-core/test/Generators.hs

```diff
@@ -18,39 +18,40 @@ import qualified Hedgehog.Range as Range
 
 import Data.Core
 import Data.Name
+import Data.Term
 
 -- The 'prune' call here ensures that we don't spend all our time just generating
 -- fresh names for variables, since the length of variable names is not an
 -- interesting property as they parse regardless.
-name :: MonadGen m => m Name
-name = Gen.prune (User <$> names) where
+name :: MonadGen m => m (Named User)
```

Comment:
Parsed/pretty-printed `Core` should _never_ contain `Gensym`s, so we simply disallow them by using `User` instead of `Name`.